src/components/SelectionPills/SelectionPill/SelectionPill.js

```diff
@@ -0,0 +1,150 @@
+
+    /** Any additional props to add to the list element (e.g. data attributes). */
+    listElementAttributes: PropTypes.object,
+
+    /** Any additional props to add to the checkbox element (e.g. data attributes). */
+    elementAttributes: PropTypes.object,
+
+    /** Flag determining if component selected state is controlled by parent through props or internal state */
+    parentControlledState: PropTypes.bool,
+
+    /** Determines wether or not selected styles are applied and start is a selected state */
+    isSelected: PropTypes.bool,
+
+    /** Determines Whether or not the pill is disabled. */
+    isDisabled: PropTypes.bool,
+
+    /** Callback function called after pill click
+     * @param {SyntheticEvent} event The react `SyntheticEvent`
+     * @param {props} object All the props passed to the component
+     */
+    onClick: PropTypes.func,
+
+    /** Callback function called after pill gained focus
+     * @param {SyntheticEvent} event The react `SyntheticEvent`
+     * @param {props} object All the props passed to the component
+     */
+    onFocus: PropTypes.func,
+
+    /** Callback function called after pill has lost focus
+     * @param {SyntheticEvent} event The react `SyntheticEvent`
+     * @param {props} object All the props passed to the component
+     */
+    onBlur: PropTypes.func,
+
+    /** Snacks theme attributes provided by Themer */
+    snacksTheme: themePropTypes,
+
+    /** Text to appear inside pill */
+    text: PropTypes.string,
+
+    /** Optional style overrides for button and its states */
+    style: PropTypes.shape({
+      button: PropTypes.object,
+      disabledStyle: PropTypes.object,
+      selectedStyle: PropTypes.object,
+      focusedStyle: PropTypes.object,
+    }),
+
+    /** Aria overrides for accessibility (i.e. use if label is not descriptive enough for screen readers) */
+    aria: PropTypes.shape({
+      label: PropTypes.string,
+    }),
+  }
+
+  static defaultProps = {
+    elementAttributes: {},
+    isSelected: false,
+    parentControlledState: false,
+    onClick: NoOp,
+    onFocus: NoOp,
+    onBlur: NoOp,
+    style: {},
+    aria: {},
+  }
+
+  state = {
+    isSelected: this.props.isSelected,
+    isFocused: false,
+  }
+
+  handleChange = event => {
+    const { onClick } = this.props
+    const { isSelected } = this.state
+
+    this.setState({ isSelected: !isSelected })
+    onClick(event, { ...this.props, isSelected: !isSelected })
+  }
+
+  toggleFocus = () => {
+    const { isFocused } = this.state
+    this.setState({ isFocused: !isFocused })
+  }
+
+  handleFocus = event => {
+    this.toggleFocus()
+    this.props.onFocus(event, { ...this.props, isFocused: !this.state.isFocused })
+  }
+
+  handleBlur = event => {
+    this.toggleFocus()
+    this.props.onBlur(event, { ...this.props, isFocused: !this.state.isFocused })
+  }
+
+  renderInputBtn({ selected }) {
+    const { aria, isDisabled, id } = this.props
+    const componentStyles = getStyles({})
```

Comment:
small style nit - feels better to have `{}` set as the default argument for `getStyles`.